### ISSUE TITLE
cannot modify kind via filters in gmake2 if kind is None

### ISSUE THREAD

**Issue by Dahrkael** [premake/premake-core#1839]
**What seems to be the problem?**
it is not possible to override the 'kind' of a gmake2 project if its set as None by default.

**What did you expect to happen?**
it should be possibly to modify all properties including 'kind' via filters.

**What have you tried so far?**
as a workaround, using normal Lua "if" it is possible to modify the kind

**How can we reproduce this?**
```lua
workspace "example"
  configurations { "Debug", "Release" }

  project "fine"
    kind "ConsoleApp"
    files {  "main.cpp" }

  project "troublemaker"
    kind "None"
    filter "configurations:Debug"
      kind "ConsoleApp"
    filter  {}
    files { "main.cpp" }
```

**What version of Premake are you using?**
premake5 (Premake Build Script Generator) 5.0.0-beta1 with the relevant backported commit

**Anything else we should know?**
support for None projects was added in this commit: https://github.com/premake/premake-core/commit/42d716b3efb9fa872e3b681977827f274469d32e
it seems like projects are removed too early during workspace generation and project filters are not processed

**Comment by nickclark2016** [comment]
Agreed that this is a bug. If you set the project-wide `kind` to something other than `none`, does that work?

**Comment by Dahrkael** [comment]
expanding on the example,
switching between `kind`s that are not `none` **does work**.
switching from `none` to any other kind **does not work**.
switching from anything to `none` **does not work** either.

`workspace "example"
  location "./solutions/"
  configurations { "Debug", "Release" }

  project "fine"
    kind "ConsoleApp"
    files {  "main.cpp" }

  project "fine2"
    kind "StaticLib"
    filter "configurations:Debug"
      kind "SharedLib"
    filter  {}
    files { "main.cpp" }
	
  project "troublemaker"
    kind "None"
    filter "configurations:Debug"
      kind "ConsoleApp"
    filter  {}
    files { "main.cpp" }
	
  project "troublemaker2"
    kind "ConsoleApp"
    filter "configurations:Debug"
      kind "None"
    filter  {}
    files { "main.cpp" }`

**Comment by samsinsane** [comment]
@nickclark2016 I'm not sure this is a bug, `None` isn't defining the type of the output, it's defining the type of project. Similar to `Utility`, `Makefile` or `SharedItems`.

@Dahrkael if you want to exclude a project from building for a specific configuration use:
```lua
removeconfigurations { "Debug" }
```

[This](https://premake.github.io/docs/Configurations-and-Platforms#per-project-configurations) covers the concept for both `platforms` and `configurations`.

**Comment by nickclark2016** [comment]
Just poking on this to see if @samsinsane's response was an answer.  If so, please either close or follow up so we can look into addressing this further @Dahrkael.

**Comment by Dahrkael** [comment]
indeed it does solve the problem when excluding certain platforms or configs, but what would be the "official" approach for the opposite then? excluding all except one. is something like this expected?
```
removeconfigurations { "*" }
configurations {"Debug" }
```

either way I personally expect premake to behave the same on the high level regardless of the module used. even if the output differs it doesn't make sense (to me) that it stops processing project filters at different stages.

**Comment by nickclark2016** [comment]
There is no official approach at this time.
